src/main/java/beam/agentsim/agents/choice/logit/LatentClassChoiceModel.scala

```diff
@@ -11,34 +11,35 @@ import org.supercsv.io.CsvBeanReader
 import org.supercsv.prefs.CsvPreference
 
 import scala.beans.BeanProperty
+import scala.collection.mutable
 
 /**
   * BEAM
   */
 class LatentClassChoiceModel(override val beamServices: BeamServices) extends HasServices {
-  val lccmData: Vector[LccmData] = parseModeChoiceParams(beamServices.beamConfig.beam.agentsim.agents.modalBehaviors.lccm.paramFile)
+  val lccmData: IndexedSeq[LccmData] = parseModeChoiceParams(beamServices.beamConfig.beam.agentsim.agents.modalBehaviors.lccm.paramFile)
 
   val classMembershipModels: Map[TourType, MultinomialLogit] = extractClassMembershipModels(lccmData)
   val modeChoiceModels: Map[TourType, Map[String, MultinomialLogit]] = {
     val mods = extractModeChoiceModels(lccmData)
     mods
   }
 
-  def parseModeChoiceParams(lccmParamsFileFile: String): Vector[LccmData] = {
+  def parseModeChoiceParams(lccmParamsFileFile: String): IndexedSeq[LccmData] = {
     val beanReader = new CsvBeanReader(IOUtils.getBufferedReader(lccmParamsFileFile), CsvPreference.STANDARD_PREFERENCE)
     val header = beanReader.getHeader(true)
     val processors: Array[CellProcessor] = LatentClassChoiceModel.getProcessors
 
     var row: LccmData = new LccmData()
-    var data: Vector[LccmData] = Vector()
+    val data = mutable.ArrayBuffer[LccmData]()
     while (beanReader.read[LccmData](row, header, processors: _*) != null) {
-      if (Option(row.value).isDefined && !row.value.isNaN) data = data :+ row.clone().asInstanceOf[LccmData]
+      if (Option(row.value).isDefined && !row.value.isNaN) data += row.clone().asInstanceOf[LccmData]
       row = new LccmData() // Turns out that if beanReader encounters a missing field, it just doesn't do the setField() rather than set it to null.... so we need to mannualy reset after every read
     }
     data
```

Comment:
data is still an ArrayBuffer under the covers, and technically mutable. I'd propose `data.toIndexedSeq`

Reply:
If I get this right [TraversableOnce.scala](https://github.com/scala/scala/blob/6ac6da8b61a3a427089a166c7802a940eac71064/src/library/scala/collection/TraversableOnce.scala#L308), `toIndexedSeq` will allocate new buffer and copy elements there - more GC pressure.